lib/pantry/helpers/handler_helper.rb

```diff
@@ -3,19 +3,44 @@ module Pantry
   module Helpers
 
     class HandlerHelper
+      class << self
 
-      def self.create_response(session_attributes: {}, message: , card: nil, end_session: true)
-        response = AlexaRubykit::Response.new
-        response.add_speech message.to_s
-        session_attributes.each do |k, v|
-          response.add_session_attribute k, v
+        def create_response(session_attributes: {}, message: , card: nil, reprompt: nil, end_session: true)
+          response = AlexaRubykit::Response.new
+          response.add_speech message.to_s
+          session_attributes.each do |k, v|
+            response.add_session_attribute k, v
+          end
+          if card.present?
+            response.add_hash_card({ type: card[:type], title: card[:title], content: card[:content] })
+          end
+          if reprompt.present?
+            response.add_reprompt reprompt
+          end
+          response.build_response(end_session)
         end
-        if card.present?
-          response.add_hash_card({ type: card['type'], title: card['title'], content: card['content'] })
+
+        def prepare_items_for_message(items)
+          items.each do |item|
+            "#{item.quantity} #{item.name}"
+          end.to_sentence
+        end
+
+        def prepare_items_for_card_with_date(items)
+          list = items.reduce([]) do |array, item|
+            array << "#{item.quantity} #{item.name} purchased on #{item.created_at.strftime("%a, %B %d")}"
+          end
+          list.join("\n")
         end
-        response.build_response(end_session)
-      end
 
+        def prepare_items_for_card_without_date(items)
+          list = items.reduce([]) do |array, item|
+            array << "#{item.quantity} #{item.name}"
+          end
```

Comment:
Same here.  Any time you find yourself doing `.reduce([])` without a nil check in the following block, you can use `.map`.